### PR TITLE
implement db support

### PR DESCRIPTION
**Pull request by batyrf**

### CODE REVIEW

**Comment by batyrf** [comment]
Hi @igorwojda. Is there a way of running codacy and codefactor checks locally?

**Comment by igorwojda** [comment]
I don't think so

File: feature_album/src/main/kotlin/com/igorwojda/showcase/feature/album/data/model/AlbumDataModel.kt [hunk 1]
```diff
@@ -12,6 +13,10 @@ internal data class AlbumDataModel(
     @field:Json(name = "image") val images: List<AlbumImageDataModel>?
 )
 
+internal fun AlbumDataModel.toEntity() =
+    AlbumDataEntity(mbId = this.mbId!!, name = this.name, artist = this.artist,
+        images = this.images?.mapNotNull { it.toEntity() } ?: listOf())
+
```
Comment (batyrf) [inline feature_album/src/main/kotlin/com/igorwojda/showcase/feature/album/data/model/AlbumDataModel.kt:19]:
```suggestion
internal fun AlbumDataModel.toEntity() =
    AlbumDataEntity(
        mbId = this.mbId!!, name = this.name, artist = this.artist,
        images = this.images?.mapNotNull { it.toEntity() } ?: listOf()
    )
```

**Comment by batyrf** [comment]
Hi @igorwojda. Did you have a chance to review pr? If there are things you want me to fix or improve, please let me know. I'm willing to spend some time.

**Comment by igorwojda** [comment]
I will rearrange, the code a bit and merge this soom

**Comment by igorwojda** [comment]
I will close this PR and resolve conflicts with another. 
https://github.com/igorwojda/android-showcase/pull/149

**Pull request closed without merging**